pkg/restore/restore.go

```diff
@@ -694,6 +694,18 @@ func (ctx *context) restoreResource(resource, namespace, resourcePath string) (a
 				continue
 			}
 
+			// Check if the PV exists in the cluster before attempting to create
+			// a volume from the snapshot, in order to avoid orphaned volumes (GH #609)
+			pv, err := resourceClient.Get(name, metav1.GetOptions{})
+			if err != nil && !apierrors.IsNotFound(err) {
+				ctx.log.Errorf("Error checking for existence of PV %s: %s", name, err)
+				continue
+			}
+			if pv != nil {
+				ctx.log.Infof("PV %s already exists in cluster, not restoring from snapshot", name)
+				continue
```

Comment:
Wondering if we should `continue` here, or *just* skip restoring the snapshot.  One downside of continuing is that the restore results won't show that the PV wasn't restored because it already existed.  WDYT?

Reply:
Agreed, should probably just skip the snapshot part and keep going with the restore.

Reply:
I'll update the backport to reflect this as well.

Reply:
If I'm understanding you correctly here @skriss, you're saying remove the `continue`, and make L709-732 an `else` branch of the `if pv != nil` condition? Looking at this again, wouldn't that be functionally equivalent, since everything in this block only applies to PersistentVolumes?

Reply:
Actually, if we just call `addToResult` instead of `ctx.log.Infof`, would that solve the issue?

Reply:
if we `continue`, then L755+ don't get executed.  I don't necessarily see a reason *not* to execute them - yes, the PV will already exist, but I'd rather have that (the API object already existing) go through the same codepath as all the other resources, and keep this change *just* to the restoring of the snapshot.  Thoughts?

Reply:
I believe the latest update addresses this. The error now shows up on the restore object, just like everything else, and we only execute snapshot restore logic if there's no pv already found.